jwst/lib/set_telescope_pointing.py

```diff
@@ -1364,6 +1399,36 @@ def all_pointings(mnemonics):
     return pointings
 
 
+
+def populate_model_from_siaf(model, siaf):
+    """
+    Populate the WCS keywords of a Level1bModel from the SIAF.
+
+    Parameters
+    ----------
+    model : `~jwst.datamodels.Level1bModel`
+        Input data as Level1bModel.
+    siaf : namedtuple
+        The WCS keywords read in from the SIAF.
+    """
+    # If not an imaging mode, update the pointing only.
+    model.meta.wcsinfo.v2_ref = siaf.v2_ref
+    model.meta.wcsinfo.v3_ref = siaf.v3_ref
+    model.meta.wcsinfo.v3yangle = siaf.v3yangle
+    model.meta.wcsinfo.vparity = siaf.vparity
+    if model.meta.exposure.type.lower() in TYPES_TO_UPDATE:
+        # For imaging modes update the pointing and
+        # the FITS WCS keywords.
+        model.meta.wcsinfo.ctype1 = 'RA--TAN'
+        model.meta.wcsinfo.ctype2 = 'DEC-TAN'
+        model.meta.wcsinfo.wcsaxes = 2
+        model.meta.wcsinfo.cunit1 = "deg"
+        model.meta.wcsinfo.cunit2 = "deg"
+        model.meta.wcsinfo.crpix1 = siaf.crpix1
+        model.meta.wcsinfo.crpix2 = siaf.crpix2
+        model.meta.coordinates.reference_frmae = "ICRS"
```

Comment:
"reference_frmae" needs to be "reference_frame"

Reply:
Good catch! Fixed now.

Reply:
Sometimes I'm amazed how many little things like that can jump out just from a quick scan with a different set of eyes. 👍